SPECS/python-pip/python-pip.spec

```diff
@@ -0,0 +1,44 @@
+%global srcname pip
+%global _description %{expand:
+A tool for installing and managing Python packages}
+
+Summary:        A tool for installing and managing Python packages
+Name:           python-pip
+Version:        24.0
+Release:        1%{?dist}
+License:        MIT AND Python-2.0.1 AND Apache-2.0 AND BSD-2-Clause AND BSD-3-Clause AND ISC AND LGPL-2.1-only AND MPL-2.0 AND (Apache-2.0 OR BSD-2-Clause)
+Vendor:         Microsoft Corporation
+Distribution:   Azure Linux
+Group:          Development/Tools
+URL:            https://pip.pypa.io/
+Source0:        https://github.com/pypa/pip/archive/%{version}/%{srcname}-%{version}.tar.gz
+
+BuildArch:      noarch
+
+%description    %{_description}
+
+%package -n python3-pip
+Summary:        %{summary}
+BuildRequires:  python3-devel
+
+%description -n python3-pip %{_description}
+
+%prep
+%autosetup -n %{srcname}-%{version}
+
+%build
+%py3_build_wheel
+
+%install
+pip3 install --no-cache-dir --no-index --ignore-installed --root %{buildroot} \
```

Comment:
does this work? Using `pip3` to install pip into the buildroot? Since this package is what actually provides `pip3` I'm surprised this works (without having to create a "circular dependency")...

I would expect something like `%{python3} dist/%{python_wheel_name}/pip install ...` to use the in-package pip instead of the (non-existent?) system pip...

Reply:
It does work in the toolchain build environment, since the python3.12 installation installs `pip3`. It will probably  not work in the buddybuild/pkggen build environment, though. Let me see if I can modify this to build in both environments properly.

Reply:
I'm also curious how this worked. The 2.0 version of Python seems not to provide any binary/script/file in general called `pip3` or `pip`. It's only available from the `python3-pip` packages, which we're removing right now. I think this may work in the toolchain build, if the raw toolchain installs `pip` but let's also make it work for a regular build. @ddstreetmicrosoft's suggestion to run the `pip` we just built sounds very reasonable to me.

Reply:
I've fixed the building of this package in buddybuild/pkggen environment by adding ensurepip to install pip3, and adding a BR on python3-wheel. Note that this BR is temporarily disabled as it will cause a pkg build error until python3-wheel is moved into the toolchain. (toolchain packages cannot build require non-toolchain packages).